chrome/content/zotero/zoteroPane.js

```diff
@@ -2187,6 +2191,185 @@ var ZoteroPane = new function()
 	});
 	
 	
+	this.sync = function () {
+		let syncReminder = document.getElementById('sync-reminder-container');
+		if (!syncReminder.collapsed) {
+			syncReminder.collapsed = true;
+		}
+
+		Zotero.Sync.Server.canAutoResetClient = true;
+		Zotero.Sync.Server.manualSyncRequired = false;
+		Zotero.Sync.Runner.sync();
+	};
+
+
+	this._syncRemindersObserverID = null;
+	this.setupSyncReminders = function (startup) {
+		if (startup) {
+			Zotero.Notifier.registerObserver(
+				{ notify: (event) => {
+					// When the API Key is deleted we need to add an observer
+					if (event === 'delete') {
+						Zotero.Prefs.set('sync.reminder.setup.enabled', true);
+						Zotero.Prefs.set('sync.reminder.setup.lastDisplayed', String(Date.now()));
+						ZoteroPane.setupSyncReminders(false);
+					}
+					// When API Key is added we can remove the observer
+					else if (event === 'add') {
+						ZoteroPane.setupSyncReminders(false);
+					}
+				} },
+				'api-key');
+		}
+
+		// If both reminders are disabled, we don't need an observer
+		if (!Zotero.Prefs.get('sync.reminder.setup.enabled')
+			&& !Zotero.Prefs.get('sync.reminder.autoSync.enabled')) {
+			if (this._syncRemindersObserverID) {
+				Zotero.Notifier.unregisterObserver(this._syncRemindersObserverID);
+				this._syncRemindersObserverID = null;
+			}
+			return;
+		}
+
+		// If we are syncing and auto-syncing then no need for observer
+		if (Zotero.Sync.Runner.enabled && Zotero.Prefs.get('sync.autoSync')) {
+			if (this._syncRemindersObserverID) {
+				Zotero.Notifier.unregisterObserver(this._syncRemindersObserverID);
+				this._syncRemindersObserverID = null;
+			}
+			return;
+		}
+
+		// If we already have an observer don't add another one
+		if (this._syncRemindersObserverID) {
+			return;
+		}
+
+		const eventTypes = ['add', 'modify', 'delete'];
+		this._syncRemindersObserverID = Zotero.Notifier.registerObserver(
+			{ notify: (event) => {
+				if (!eventTypes.includes(event)) {
+					return;
+				}
+				setTimeout(() => {
+					this.showSetupSyncReminder();
+					this.showAutoSyncOffReminder();
+				}, 5000);
+			} },
+			'item',
+			'syncReminder');
+	};
+
+
+	this.showAutoSyncOffReminder = function () {
+		const sevenDays = 1000 * 60 * 60 * 24 * 7;
+
+		// Reasons not to show reminder:
+		// - User turned reminder off
+		// - Sync is not enabled
+		// - Auto-Sync is enabled
+		// - Last sync for all libraries was within 7 days
+		if (!Zotero.Prefs.get('sync.reminder.autoSync.enabled')
+			|| !Zotero.Sync.Runner.enabled
+			|| Zotero.Prefs.get('sync.autoSync')
+			|| !Zotero.Libraries.getAll()
+				.find(library => library.lastSync.getTime() < (Date.now() - sevenDays))) {
```

Comment:
1) Need to `filter` on `library.syncable`, since otherwise this will throw if there's a feed library
2) Clearer to switch around comparison and use `every()` with no negation